packages/block-editor/src/store/private-selectors.js

```diff
@@ -53,25 +57,17 @@ export function getLastInsertedBlocksClientIds( state ) {
  *
  * @return {boolean} Whether the block descendants are disabled.
  */
-export const isBlockSubtreeDisabled = createSelector(
-	( state, clientId ) => {
-		const isChildSubtreeDisabled = ( childClientId ) => {
-			return (
-				getBlockEditingMode( state, childClientId ) === 'disabled' &&
-				getBlockOrder( state, childClientId ).every(
-					isChildSubtreeDisabled
-				)
-			);
-		};
-		return getBlockOrder( state, clientId ).every( isChildSubtreeDisabled );
-	},
-	( state ) => [
-		state.blocks.parents,
-		state.blocks.order,
-		state.blockEditingModes,
-		state.blockListSettings,
-	]
-);
+export const isBlockSubtreeDisabled = ( state, clientId ) => {
+	const isChildSubtreeDisabled = ( childClientId ) => {
+		return (
+			getBlockEditingMode( state, childClientId ) === 'disabled' &&
+			getBlockOrder( state, childClientId ).every(
+				isChildSubtreeDisabled
+			)
+		);
+	};
+	return getBlockOrder( state, clientId ).every( isChildSubtreeDisabled );
+};
```

Comment:
After #58349, this is no longer called, so the performance improvement of this PR will be lessened.

Reply:
Probably still worth to remove the memoization I think... Also the dependencies of this selector weren't fully correct.

Reply:
Also, this selector should also check the editing mode of the top-level `clientId`, not just its children.

If I look at the only call in `BlockListBlockProvider`, I see that it's done anyway, checking `blockEditingMode === 'disabled' && isBlockSubtreeDisabled()`.